include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -115,7 +115,7 @@ class kmer_hash_view : public std::ranges::view_interface<kmer_hash_view<urng_t>
      */
     auto begin() noexcept
     {
-        return shape_iterator<urng_t>{std::ranges::begin(urange), shape_};
+        return shape_iterator<urng_t>{std::ranges::begin(urange), std::ranges::end(urange), shape_};
```

Comment:
I think it would require fewer changes if you just check `std::distance` of `std::ranges::begin(urange), std::ranges::end(urange)` here and `return std::ranges::end(urange)` if the requirement is not met instead of passing an extra argument to the constructor?

Reply:
No, because that would be a different type than the shape_iterator, which first is wrong and second not possible due to different types in the return expression deduced at runtime. The iterator must know whether it is at the end or not, so that `size()`  works correctly. For this it has to set the `text_right` iterator correctly (in this case to the end of the underlying range). So, you can't do it without an extra argument to the constructor.